lib/users.js

```diff
@@ -504,6 +504,11 @@ Users.prototype.enrichProfile = function (profile, noGroups, callback) {
 
 Users.prototype._getAllGroupsADCached = function (profile, callback) {
   var self = this;
+  
+  if (!self._groupsCache) {
```

Comment:
This exit is needed, otherwise, the `_getAllGroupsADCached` method fails when it attempts to access the cache [here](https://github.com/auth0/ad-ldap-connector/blob/a4806073063f5fddb3fe978a6336e96abd25c1a2/lib/users.js#L516).